STM32F042K4Tx/Core/Src/main.c

```diff
@@ -114,29 +121,49 @@ int main(void)
   /* Initialize all configured peripherals */
   MX_GPIO_Init();
   MX_CAN_Init();
-  MX_TIM3_Init();
-  MX_TIM14_Init();
+  MX_TIM16_Init();
   /* USER CODE BEGIN 2 */
-  HAL_CAN_Start(&hcan);
-  HAL_TIM_Base_Start_IT(&htim14);
-  __HAL_TIM_SET_COUNTER(&htim14, 0);
+	HAL_CAN_Start(&hcan);
+	if(CANBus_init(&hcan, &htim16)!= HAL_OK)

```

Comment:
Why are you using the same timer for the can bus init and for the encoder?

Reply:
This is a good question... I'm not exactly sure why this is, but I'll separate the CAN and encoder into two different timers.